### DISCUSSION TITLE
NRF52840DK with sx1262mb2xAS shield  - MCPS CONFIRM status: Rx 2 timeout

### DISCUSSION THREAD

**Discussion by carriegong** [Lora-net/LoRaMac-node#1123] [Q&A]
Are there any solutions here for getting an error : MCPS CONFIRM status: Rx 2 timeout for confirmed messages. 
It is missing downlinks too when sending and receiving at the same time. 

Unconfirmed messages seems to be ok. but they never receive downlinks.   
Class A, downlinks on port 1

**Comment by rfenderStrata** [comment]
Hello cawahe we are currently trying to adapt LoRaWAN for NRF52840 too, did you get your error from your own implementation of Lora-net/LoRaMac-node or are you using  this repo [nrf52-lora](https://github.com/gluedig/nrf52-lora) 
?

**Comment by carriegong** [comment]
> Hello cawahe we are currently trying to adapt LoRaWAN for NRF52840 too, did you get your error from your own implementation of Lora-net/LoRaMac-node or are you using this repo [nrf52-lora](https://github.com/gluedig/nrf52-lora)
> ?

From this example. [https://github.com/FreeRTOS/Lab-Project-FreeRTOS-LoRaWAN](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-LoRaWAN),  but seems like sending sends on all freq even after join. How can I alter that?

**Reply by zzolax** [reply]
Hi,
carriegong did you already find solution for this? Have similar problem.

**Reply by brian-level** [reply]
I am also seeing this issue.  I found a temporary workaround by forcing a 5ms delay between each SPI write of an opcode to the radio.  Its not 100% reliable, and still seems to only work in class C
`
void SX126xWriteCommand(RadioCommands_t opcode, uint8_t *buffer, uint16_t size)
{
	uint8_t req[] = {
		opcode
	};

	if (opcode == 0x82)
		togglesignal(400);

	LOG_DBG("Issuing opcode 0x%x w. %" PRIu16 " bytes of data",
		opcode, size);
	//togglesignal(10);
	sx126x_spi_transceive(req, NULL, sizeof(req), buffer, NULL, size);

	/* HORRIBLE HACK --- I added this after many hours of debugging to find that
	 * this delay here allows ack packets to be rx'd.  Without this hack the
	 * Rx interrupt is a "rx timeout" interrupt and rx fails
	 */
	k_busy_wait(5000);
	//togglesignal(30);
}
`

**Reply by carriegong** [reply]
you may look at https://github.com/FreeRTOS/Lab-Project-FreeRTOS-LoRaWAN/issues/3#issuecomment-1015698495
Maybe this will solve the problem?

**Reply by zzolax** [reply]
Thanks but it didn't. In OTAA all is ok, just in ABP have this issue (node recives NewChannelReq for every message). Maybe you know any working ABP Lora library for NRF52840 ?
